### DISCUSSION TITLE
Rule condition before trigger

### DISCUSSION THREAD

**Discussion by mczac** [arendst/Tasmota#12182] [Q&A] [answered]
Hi there, I couldn't figure out if there is a way to have a rule condition that gets executed before trigger change was effectuated. I have Sonoff 4CH. 2 Relays are Interlocked. Third Relay R3 needs to be triggered just before any of R1 or R2 goes on. and R3 will need to go off as soon as R1 or R2 goes off.

**Comment by ascillato2** [comment, marked as answer]
Hi,

To achieve that, you should make a rule and a delay between commands. Tasmota can't make a pretrigger trigger.

If you are using a physical button, you can make a rule for that button and you can use backlog inside your rule to perform that series of commands.

**Reply by mczac** [reply]
Thanks for your active responses. Appreciate it. I am controlling Sonoff 4CH via MQTT, I can structure my MQTT that first I send activate R3 and then perform an action. I am afraid that racing could happen unless I do QOS2 and wait for status change. is there a trigger in Tasmota to action a script based on received MQTT, that would work for me. I will then trigger script. thanks!

**Reply by sfromis** [reply]
You could also control it with a local rule, where you via MQTT can post one command (like the Event command) to activate a rule, which then can do the required sequence of actions, including controlling timing. And you could decouple buttons from automatically triggering relays, to pass this through the same rule logic.

**Reply by mczac** [reply]
Thanks, that would work! great idea
